.github/workflows/cli-regression.yml

```diff
@@ -0,0 +1,41 @@
+name: CLI Regression Test
+
+on:
+  pull_request:
+    types: [opened]
+  push:
+    branches:
+      - '**'
+
+jobs:
+  cli-regression-test:
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.9.7
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.9.7"
```

Comment:
Heads up that we're now running this job on Python 3.10.1 (this specific patch version because it's the newest Python 3.10 version available in the CircleCI images we're using. Using any available Python 3.10 version would probably be fine) due to the issues we ran into with Python 3.9.7. Would recommend also bumping all of these to use Python 3.10 (or anything other than 3.9.7, which will fail once the latest changes are pulled in from `develop`.

Reply:
Really appreciate the heads up, updated to 3.10!